Conditions for link up

ajoaoff opened the pull request.

ajoaoff committed:
Test both ends of link before setting it as active. Fix #75.

ajoaoff committed:
Created a timer when a link change status.

A link now is considered up only if it stays up for at least the defined time. Fix #77.

ajoaoff committed:
Removing debug message.

hdiogenes commented on main.py:
```diff
@@ -23,6 +25,7 @@ def setup(self):
         """Initialize the NApp's links list."""
         self.links = {}
         self.store_items = {}
+        self.link_up_timer = settings.LINK_UP_TIMER
```
Maybe it would be better to have a default value here? This way the code won't break if a value is missing from the configuration.

```suggestion
        self.link_up_timer = getattr(settings, 'LINK_UP_TIMER', 10)
```

ajoaoff replied:
You're right. It's done.

ajoaoff committed:
Default value for link up timer

Use getattr to get the value of the setting LINK_UP_TIMER so
a default value can be used if the setting is unset.

ajoaoff committed:
Fix linter issues

hdiogenes commented on main.py:
```diff
@@ -373,11 +376,26 @@ def handle_interface_link_up(self, event):
-            self.notify_topology_update()
-            self.update_instance_metadata(interface.link)
-            self.notify_link_status_change(link)
+            time.sleep(self.link_up_timer)
```
How can we make sure the NApp thread won't be locked for the time of this sleep?

hdiogenes commented on main.py:
```diff
@@ -23,6 +25,7 @@ def setup(self):
         """Initialize the NApp's links list."""
         self.links = {}
         self.store_items = {}
+        self.link_up_timer = getattr(settings, 'LINK_UP_TIMER', 10)
```
I know this code was my own suggestion (thanks for accepting it! :-), but now I have the feeling that this hard-coded constant should be declared as a global (`DEFAULT_LINK_UP_TIMER`, for example).

ajoaoff committed:
Fix test on handle link up

ajoaoff committed:
Fix linter issues

ajoaoff committed:
Merge branch 'master' into link_up_conditions
